app/controllers/questionings_controller.rb

```diff
@@ -63,18 +63,18 @@ def destroy
     render nothing: true, status: 204
   end
 
-  # Re-renders the fields in the condition form when requested by ajax.
+  # Responds to ajax request with json containing data needed for condition form.
   def condition_form
     if params[:questioning_id].present?
       @questioning = Questioning.find(params[:questioning_id])
     else
       # Create a dummy questioning so that the condition can look up the refable qings, etc.
       @questioning = init_qing(form_id: params[:form_id])
     end
-
-    # Create a dummy condition with the given ref qing.
-    @condition = @questioning.build_condition(ref_qing_id: params[:ref_qing_id])
-    render(partial: 'conditions/form_fields')
+    authorize! :condition_form, @questioning
+    @condition = @questioning.condition || @questioning.build_condition()
+    @condition.ref_qing_id = params[:ref_qing_id]
+    render json: ConditionViewSerializer.new(@condition), status: 200
```

Comment:
Looking really good here! Again I don't think the 200 is needed though, try without?